calculator/src/main/java/com/market/banica/calculator/service/grpc/AuroraClientSideService.java

```diff
@@ -57,12 +57,17 @@ public ItemOrderBookResponse getIngredient(String productName, String clientId)
         return auroraResponse.getItemOrderBookResponse();
     }
 
+    public AuroraServiceGrpc.AuroraServiceBlockingStub getBlockingStub() {
+
+        return blockingStub;
+    }
+
     private Aurora.AuroraResponse getAuroraResponse(String message) {
         LOGGER.debug("In getAuroraResponse private method");
 
         Aurora.AuroraRequest request = buildAuroraRequest(message);
 
-        return blockingStub.request(request);
+        return getBlockingStub().request(request);
```

Comment:
you don't need to call the getter, you may directly use blockingStub

Reply:
The getter is not required for the service but allows for a complete component integration test of the component calculator service. Spying on the instance and replacing the real stub with a mocked one allows testing the correct format of the messages for the gRPC part. Otherwise(if we mock the bean AuroraServiceGrpc.AuroraServiceBlockingStub and return predefined response), we would never know if what we compose as data and format as a gRPC request would satisfy the expected service instance. Through the InProcess library of gRPC, we could predefine expected behaviour and test the full functionality of the service using gRPC.